lib/internal/util.js

```diff
@@ -409,7 +409,10 @@ function promisify(original) {
           resolve(values[0]);
         }
       });
-      ReflectApply(original, this, args);
+      if (isPromise(ReflectApply(original, this, args))) {
```

Comment:
@anonrig just pinging to make sure we're are of an OK with the potential impact this would have on perf?